MapboxSceneKit/MapboxImageAPI.swift

```diff
@@ -236,7 +242,10 @@ public final class MapboxImageAPI: NSObject {
         }
 
         group.notify(queue: DispatchQueue.main) {
-            self.pendingFetches.removeValue(forKey: groupID)
+            self.pendingFetchesDispatchQueue.sync(flags: .barrier) { [weak self] in
+                guard let self = self else { return }
+                self.pendingFetches.removeValue(forKey: groupID)
+            }
```

Comment:
I'm not too familiar with how `sync()` and `group.notify()`, but is it possible this could cause some sort of deadlock on the main queue? From the [docs](https://developer.apple.com/documentation/dispatch/dispatchqueue):

> Attempting to synchronously execute a work item on the main queue results in deadlock.

(I used a slightly different approach for wrapping this call in #80.)

Reply:
I'm not sure either. I'll take a look at your change. It might be the way to go.

Reply:
I closed the pull request already, but feel free to reopen it.

Reply:
I cannot speak into the possibility of deadlock here but this is working great for me